javascript/extractor/src/com/semmle/js/ast/jsx/JSXThisExpr.java

```diff
@@ -0,0 +1,21 @@
+package com.semmle.js.ast.jsx;
+
+import com.semmle.js.ast.ThisExpression;
+import com.semmle.js.ast.SourceLocation;
+import com.semmle.js.ast.Visitor;
+
+public class JSXThisExpr extends ThisExpression implements IJSXName {
```

Comment:
Is this class just here to create a `ThisExpression` that's also a `IJSXName`? If so, just make `ThisExpression` implement the `IJSXName` interface instead. That's what we do for `ITypeExpression`.

Reply:
I did it like this to be consistent with the other classes that `implements IJSXName`, like: `public class JSXIdentifier extends Identifier implements IJSXName`.